crates/router/src/services/api/generic_link_response.rs

```diff
@@ -40,6 +46,67 @@ pub fn build_generic_expired_link_html(
     }
 }
 
+pub fn build_payout_link_html(
+    link_data: &GenericLinkFormData,
+) -> CustomResult<String, errors::ApiErrorResponse> {
+    let mut tera = Tera::default();
+    let mut context = Context::new();
+
+    // Insert dynamic context in CSS
+    let css_dynamic_context = "{{ color_scheme }}";
+    let css_template =
+        include_str!("../../core/generic_link/payout_link/initiate/styles.css").to_string();
+    let final_css = format!("{}\n{}", css_dynamic_context, css_template);
+    let _ = tera.add_raw_template("payout_link_styles", &final_css);
+    context.insert("color_scheme", &link_data.css_data);
+
+    let css_style_tag = match tera.render("payout_link_styles", &context) {
+        Ok(css) => format!("<style>{}</style>", css),
+        Err(tera_error) => {
+            logger::error!("{tera_error}");
+            Err(errors::ApiErrorResponse::InternalServerError)?
```

Comment:
Use `attach_printable()` or `attach_printable_lazy()` to add context on what failed everywhere internal server errors are raised.

Honestly, just use `change_context()` instead of logging the `tera` error, and add context using `attach_printable()` / `attach_printable_lazy()`.